subsystem/claw.py

```diff
@@ -0,0 +1,62 @@
+import math
+from robotpy_toolkit_7407 import Subsystem
+from robotpy_toolkit_7407.motors.rev_motors import SparkMax, SparkMaxConfig
+from robotpy_toolkit_7407.sensors.limit_switches.limit_switch import LimitSwitch
+from robotpy_toolkit_7407.subsystem_templates.drivetrain import SwerveDrivetrain
+import config
+import ctre
+import constants
+import wpilib
+
+class Claw(Subsystem):
+
+    claw_motor: SparkMax = SparkMax(config.claw_motor_extend_id)
```

Comment:
Same comment as [here](https://github.com/Choate-Robotics/7407-DriveCode-ChargedUp/pull/9#discussion_r1083373806) about the value of initializing dependencies outside of the class that is using them.

Reply:
Good job adding the params to the constructor. There's a side effect I don't think you realize which requires a bit more change, but it's easy.

TLDR: Remove constructing the spark and piston to robot.py or whichever file you setup this subsystem.

Checkout [this example](https://gist.github.com/mllewellyn/043c61e75dce4d5981fc42d7c21e17e2) I made. See how despite us passing in new objects, the internal ones are still setup (and then thrown away)?

This is undesirable because if there's a side effect to constructing a `SparkMax` (such as a CAN ID being reserved in some system, which I think there is) when multiple tests run one after the other a good test can break because an unrelated test had an unexpected side effect.

Reply:
Also type hints in the construct would be nice :-) 

Long term we should setup something like mypy to automatically tell us when we need to type hint so that it's not a bother on reviews.

Reply:
I see what you are trying to say, thanks for the example. I reworked the class so that the motor is passed through the constructor (so no overrides will occur). I also marked its type as a SparkMax so other functions would still show.

Let me know if this is what you were looking for.

Reply:
Exactly what I was looking for with the spark in the claw subsystem. Don't forget to setup the spark max in robot.py and pass it in though! Or forget about the `SingleSolenoidPiston` :-)